src/main/java/mods/eln/misc/Obj3DFolder.java

```diff
@@ -2,59 +2,60 @@
 
 import mods.eln.misc.Obj3D.Obj3DPart;
 
+import java.io.IOException;
+import java.net.URLDecoder;
+import java.security.CodeSource;
+import java.util.Enumeration;
 import java.util.HashMap;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
 
+/**
+ * Utility class used to load all eln models and corresponding obj files.
+ */
 public class Obj3DFolder {
 
-    HashMap<String, Obj3D> nameToObjHash = new HashMap<String, Obj3D>();
+  private HashMap<String, Obj3D> nameToObjHash = new HashMap<String, Obj3D>();
 
-	/* public void loadFolder(String modName, String folderName) {
-
-			URI url;
-			try {
-				URL rec = mods.eln.Eln.class.getResource("/assets/" + modName +  folderName);
-				
-				if(rec == null) {
-					Utils.println("if(rec == null)");
-				} else {
-					url = rec.toURI();
-					if (url == null) {
-						Utils.println("if(url == null)");
-					} else {
-					    File dir = new File(url);
-						//File dir = new File(mods.eln.Eln.class.getResource(folderName).getFile());
-					    File[] lol = dir.listFiles();
-					    for (File file : dir.listFiles()) {
-						    if (file.isFile()) {
-						    	if(file.getName().endsWith(".obj")||file.getName().endsWith(".OBJ")) {
-							    //	String fileName = folder + file.getName();
-							    	Obj3D obj =  new Obj3D();
-							    	obj.loadFile(modName,folderName + "/" + file.getName());
-							    	//Utils.println( '"' + folderName + "/" + file.getName() + '"' + ',');
-							    	String tag = file.getName().replaceAll(".obj", "").replaceAll(".OBJ", "");
-							    	nameToObjHash.put(tag,obj);
-						    	}
-						    }
-						    if(file.isDirectory()) {
-						    	//String bi = file.getName();
-						    	loadFolder(modName,folderName + "/" + file.getName());
-						    }
-						}	
-					}
-				}
-			} catch (URISyntaxException e) {
-				e.printStackTrace();
-			}
-		}*/
+  /** Load all obj models available in the release mod asset folder. */
+  public void loadAllElnModels() {
```

Comment:
The debug output is available here: http://pastebin.com/36fPVW47
 * Many errors are reported because a lot models do not have txt files
 * 101 models are loaded, even if they are not used in the game...